### PR TITLE
C++: Test C++20 implicit array sizes.

### PR DESCRIPTION
**Pull request by calumgrant**
Adds tests for  C++ 20 implicit new array sizes. Also reviewed the QL libraries and did not find any use of `NewArrayExpr.getExtent()` that would benefit from using `getArraySize()` instead.

### CODE REVIEW

**Comment by jketema** [comment]
Do we use `getAllocatedType().(ArrayType).getArraySize()` directly in places where we should actually be using the predicate introduced here?

**Comment by calumgrant** [comment]
> Do we use `getAllocatedType().(ArrayType).getArraySize()` directly in places where we should actually be using the predicate introduced here?

I can't find any.

File: cpp/ql/src/change-notes/2024-07-10-newarrayexpr-arraysize.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (jketema) [inline cpp/ql/src/change-notes/2024-07-10-newarrayexpr-arraysize.md:1]: The change not should go in `cpp/ql/lib/change-notes` (it's a library change, not a query change). That should also fix the error you're getting now.

**Comment by jketema** [comment]
> > Do we use `getAllocatedType().(ArrayType).getArraySize()` directly in places where we should actually be using the predicate introduced here?
> 
> I can't find any.

I'm finding `cpp/leap-year/unsafe-array-for-days-of-the-year` and `cpp/bad-strncpy-size`. However, it's better to create an issue for follow-up work then to possibly change those queries as part of this PR.

**Review by jketema — APPROVED** [review]
LGTM.

**Pull request merged**